twitter/src/main/java/com/hazelcast/jet/contrib/twitter/TwitterSources.java

```diff
@@ -0,0 +1,390 @@
+                            .fillBufferFn(TwitterStreamSourceContext::fillTimestampedBuffer)
+                            .destroyFn(TwitterStreamSourceContext::close)
+                            .build();
+    }
+
+    /**
+     * Creates a {@link BatchSource} which emits tweets in the form of {@link
+     * Status} by using Twitter's Search API for data ingestion. Twitter
+     * restricts the repeated (continuous) access to its search endpoint so you
+     * can only make 180 calls every 15 minutes. This source tries to get the
+     * search results from the search endpoint until the API rate limit gets
+     * exhausted.
+     * <p>
+     * Example usage:
+     * <pre>{@code
+     * Properties credentials = loadTwitterCredentials();
+     * BatchSource<Status> twitterSearchSource =
+     *              TwitterSources.search(
+     *                      credentials,
+     *                      "Jet flies"
+     *              );
+     * Pipeline p = Pipeline.create();
+     * BatchStage<Status> srcStage = p.readFrom(twitterSearchSource);
+     * }</pre>
+     *
+     * @param credentials a Twitter OAuth1 credentials that consists of
+     *                    "consumerKey", "consumerSecret", "token",
+     *                    "tokenSecret" keys.
+     * @param query       a search query
+     * @return a batch source to use in {@link com.hazelcast.jet.pipeline.Pipeline#readFrom}
+     * @see <a href="https://developer.twitter.com/en/docs/basics/rate-limiting">Twitter's Rate Limiting.</a>
+     * @see <a href="https://developer.twitter.com/en/docs/tweets/search/api-reference/get-search-tweets">
+     * GET search tweets/ Twitter Developers</a>
+     * @see <a href="https://developer.twitter.com/en/docs/tweets/search/guides/standard-operators">
+     * Twitter API / Standard search Operators</a>
+     */
+    @Nonnull
+    public static BatchSource<Status> search(
+            @Nonnull Properties credentials,
+            @Nonnull String query
+    ) {
+        return SourceBuilder.batch("twitter-search-batch-source",
+                ctx -> new TwitterBatchSourceContext(credentials, query))
+                            .fillBufferFn(TwitterBatchSourceContext::fillBuffer)
+                            .build();
+    }
+
+    private static void checkTwitterCredentials(Properties credentials) {
+        String consumerKey = credentials.getProperty("consumerKey");
+        String consumerSecret = credentials.getProperty("consumerSecret");
+        String token = credentials.getProperty("token");
+        String tokenSecret = credentials.getProperty("tokenSecret");
+
+        isMissing(consumerKey, "consumerKey");
+        isMissing(consumerSecret, "consumerSecret");
+        isMissing(token, "token");
+        isMissing(tokenSecret, "tokenSecret");
+    }
+
+    private static void isMissing(String key, String description) {
+        Objects.requireNonNull(key, description);
+        if ("REPLACE_THIS".equals(key)) {
```

Comment:
Maybe this is superfluous. We never provide such a value. The file `twitter-security.properties` is actually never read, it's in tests and we don't use it in tests. We can delete it I think.

Reply:
I removed this redundant check.